src/util.ts

```diff
@@ -479,6 +480,64 @@ export function thisExtensionPackage(): PackageJSON {
     };
 }
 
+export async function GetExtensionLocalizedPackageJson(): Promise<{[key: string]: any}> {
+    let localizedFilePath: string = path.join(thisExtensionPath(), `package.nls.${getLocaleId()}.json`);
+    const fileExists: boolean = await checkFileExists(localizedFilePath);
+    if (!fileExists) {
+        localizedFilePath = path.join(thisExtensionPath(), "package.nls.json");
+    }
+    const localizedStrings = fs.readFileSync(localizedFilePath, "utf8");
+    return JSON.parse(localizedStrings);
+}
+
+interface CommandPalette {
+    command: string;
+    when: string | null;
+}
+
+function evaluateExpression(expression: string | null, context: {[key: string]: any}): boolean {
```

Comment:
Cool! Should we have unit tests for this to ensure that we're using this correctly to make sure we always understand the context right? 

Or, did you happen to pull this from the vscode core implementation, so we know that it's right? IF so, we should leave a comment with a link.

Reply:
Did not take this from vscode core implementation. Doing some research, theirs is a lot more robust I found the files that help vscode with this logic. It's very robust and is overkill for ours currently, but will be including the recycled file in case we ever include more complex conditions in our package.json when clauses.